### PR TITLE
Test controller as nuget package

### PR DESCRIPTION
**Pull request by vladimir-shcherbakov**
## Description

Change TestController implementation.
Issue #6180

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources.Test/packages.config [hunk 1]
```diff
@@ -11,17 +11,18 @@
...
   <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.2.9-preview" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.5.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.3.0" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.7.2" targetFramework="net452" />
   <package id="Microsoft.WindowsAzure.Management" version="4.1.1" targetFramework="net45" />
   <package id="Moq" version="4.2.1510.2205" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Resources/Commands.Resources.Test/packages.config]: I'm not sure these packages.config changes actually affect anything. These packages should be loaded from restoring Profile.

File: src/ResourceManager/Resources/Commands.Resources/packages.config [hunk 2]
```diff
@@ -3,4 +3,5 @@
   <package id="Microsoft.Azure.Management.Authorization" version="2.8.0-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.ManagementGroups" version="1.1.0-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.ResourceManager" version="1.9.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.PowerShell.TestFx" version="1.0.137-preview" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Resources/Commands.Resources/packages.config]: This should be added to the Profile packages.config.

File: src/ResourceManager/Storage/Commands.Management.Storage.Test/packages.config [hunk 3]
```diff
@@ -17,9 +17,9 @@
   <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.2.9-preview" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.5.0-preview" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.3.0" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.7.2" targetFramework="net452" />
+  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Storage/Commands.Management.Storage.Test/packages.config]: Same here with packages.config changes should be in Profile.

File: src/ResourceManager/Storage/Commands.Management.Storage/packages.config [hunk 4]
```diff
@@ -2,6 +2,7 @@
 <packages>
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Storage" version="8.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.PowerShell.TestFx" version="1.0.137-preview" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Storage/Commands.Management.Storage/packages.config]: Same here with packages.config changes should be in Profile.

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.cs [hunk 5]
```diff
@@ -88,7 +87,8 @@ public void TestGetADGroupWithSearchString()
         [Trait(Category.AcceptanceType, Category.CheckIn)]
         public void TestGetADGroupWithBadSearchString()
         {
-            ResourcesController.NewInstance.RunPsTest(interceptor, "Test-GetADGroupWithBadSearchString");
+            //ResourcesController.NewInstance.RunPsTest("Test-GetADGroupWithBadSearchString");
```
Comment (MiYanni) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.cs]: Commented code.

File: src/ResourceManager/Resources/Commands.Resources.Test/packages.config (hunk 1 above)
Comment (vladimir-shcherbakov) [reply src/ResourceManager/Resources/Commands.Resources.Test/packages.config]: fixed

File: tools/Common.Dependencies.Test.targets [hunk 6]
```diff
@@ -64,6 +64,10 @@
       <HintPath>$(LibraryNugetPackageFolder)\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.7.2\lib\net452\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Azure.PowerShell.TestFx, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>$(LibraryNugetPackageFolder)\Microsoft.Azure.PowerShell.TestFx.1.0.137-preview\lib\net452\Microsoft.Azure.PowerShell.TestFx.dll</HintPath>
```
Comment (markcowl) [inline tools/Common.Dependencies.Test.targets]: Can we make this one of the published packages from the common repo?

**Comment by vladimir-shcherbakov** [comment]
I'll update the guide as soon as the review has signed off to be sure there won't be any changes.

**Comment by vladimir-shcherbakov** [comment]
I removed ProfileController as redundant, but 2 others are still in use: ResourcesController uses library calls to setup and cleanup an environment like [this](https://github.com/Azure/azure-powershell/blob/preview/src/StackAdmin/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.cs#L44):
```c#
ADGroup newGroup = null;
var controllerAdmin = ResourcesController.NewInstance;

controllerAdmin.RunPsTestWorkflow(
	interceptor,
	// scriptBuilder
	() =>
	{
		newGroup = CreateNewAdGroup(controllerAdmin);
		return new[] { scriptMethod };
	},
	// cleanup
	() =>
	{
		DeleteAdGroup(controllerAdmin, newGroup);
	},
	MethodBase.GetCurrentMethod().ReflectedType?.ToString(),
	MethodBase.GetCurrentMethod().Name);
}
```
and Storage TestController is still in use for blob containers tests because to get the new one  work the tests need to be re-recorded but they use some experimental environment I don't have access to.

**Comment by markcowl** [comment]
@vladimir-shcherbakov Can you update this PR for the new released common libraries, pull the latest and fix merge conflicts, and clean up the commits?

**Comment by cormacpayne** [comment]
@vladimir-shcherbakov re-targeted the PR at the new default branch, `master`, so you will need to pull from that branch to resolve the merge conflicts rather than `preview`

**Comment by cormacpayne** [comment]
@vladimir-shcherbakov ping on the above comments

File: src/ResourceManager/Storage/Commands.Management.Storage.Test/ScenarioTests/StorageTestRunner.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            TestRunner = TestManager.CreateInstance(output)
+                .WithNewPsScriptFilename($"{GetType().Name}.ps1")
+                .WithProjectSubfolderForTests("ScenarioTests")
+                .WithCommonPsScripts(new[]
+                {
+                    @"Common.ps1",
+                })
+                .WithExtraRmModules(helper => new[]
+                {
+#if !NETSTANDARD
```
Comment (markcowl) [inline src/ResourceManager/Storage/Commands.Management.Storage.Test/ScenarioTests/StorageTestRunner.cs]: We can remove this IFDEF, now

**Pull request merged**